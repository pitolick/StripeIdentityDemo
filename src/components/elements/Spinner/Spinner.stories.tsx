import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { Spinner } from './Spinner'

const meta: Meta<typeof Spinner> = {
  title: 'elements/Spinner',
  component: Spinner,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof Spinner>

/**
 * デフォルト
 */
export const Default: Story = {
  args: {},
}

/**
 * サイズ：sm
 */
export const Sm: Story = {
  args: {
    size: 'sm',
  },
}

/**
 * サイズ：md
 */
export const Md: Story = {
  args: {
    size: 'md',
  },
}

/**
 * サイズ：lg
 */
export const Lg: Story = {
  args: {
    size: 'lg',
  },
}
