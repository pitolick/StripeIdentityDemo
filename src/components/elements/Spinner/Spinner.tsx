/**
 * スピナー
 * @module Spinner
 */
import React from 'react'

/**
 * スピナーのプロパティ
 */
interface SpinnerProps {
  /**
   * サイズ
   */
  size?: 'sm' | 'md' | 'lg'
}

/**
 * スピナーコンポーネント
 */
export const Spinner: React.FC<SpinnerProps> = ({ size = 'md' }) => {
  // サイズによって表示を変える
  let sizeClass = ''
  switch (size) {
    case 'sm':
      sizeClass = 'w-4 h-4'
      break
    case 'md':
      sizeClass = 'w-6 h-6'
      break
    case 'lg':
      sizeClass = 'w-8 h-8'
      break
  }

  return (
    <div className={`inline-block animate-spin ${sizeClass} rounded-full border-[3px] border-current border-t-transparent text-blue`} role="status" aria-label="loading">
      <span className="sr-only">Loading...</span>
    </div>
  )
}
