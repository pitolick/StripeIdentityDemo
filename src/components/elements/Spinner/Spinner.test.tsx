import React from 'react'
import { render } from '@testing-library/react'
import { Spinner } from './Spinner'

describe('Spinnerコンポーネント', () => {
  it('デフォルトのサイズで正しくレンダリングされること', () => {
    const { container } = render(<Spinner />)
    expect(container).toMatchSnapshot()
  })

  it('サイズがsmの場合、正しくレンダリングされること', () => {
    const { container } = render(<Spinner size="sm" />)
    expect(container).toMatchSnapshot()
  })

  it('サイズがmdの場合、正しくレンダリングされること', () => {
    const { container } = render(<Spinner size="md" />)
    expect(container).toMatchSnapshot()
  })

  it('サイズがlgの場合、正しくレンダリングされること', () => {
    const { container } = render(<Spinner size="lg" />)
    expect(container).toMatchSnapshot()
  })
})
