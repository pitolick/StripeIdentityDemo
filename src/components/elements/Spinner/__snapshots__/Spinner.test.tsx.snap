// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinnerコンポーネント サイズがlgの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    aria-label="loading"
    class="inline-block animate-spin w-8 h-8 rounded-full border-[3px] border-current border-t-transparent text-blue"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`Spinnerコンポーネント サイズがmdの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    aria-label="loading"
    class="inline-block animate-spin w-6 h-6 rounded-full border-[3px] border-current border-t-transparent text-blue"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`Spinnerコンポーネント サイズがsmの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    aria-label="loading"
    class="inline-block animate-spin w-4 h-4 rounded-full border-[3px] border-current border-t-transparent text-blue"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`Spinnerコンポーネント デフォルトのサイズで正しくレンダリングされること 1`] = `
<div>
  <div
    aria-label="loading"
    class="inline-block animate-spin w-6 h-6 rounded-full border-[3px] border-current border-t-transparent text-blue"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
